tests/plugins/analysis/indices/analyze/analyzer/phone.yaml

```diff
@@ -0,0 +1,37 @@
+$schema: ../../../../../../json_schemas/test_story.schema.yaml
+
+description: Test /{index}/_analyze for the phone and phone-search analyzers.
+prologues:
+  - path: /example-phone
+    method: PUT
+    request:
+      payload:
+        settings:
+          index:
+            analysis:
+              analyzer:
+                phone-ch:
+                  type: phone
+                  phone-region: CH
+                phone-search-ch:
+                  type: phone-search
+                  phone-region: CH
+epilogues:
+  - path: /example-phone
+    method: DELETE
+    status: [200, 404]
+chapters:
+  - synopsis: Analyze phone number.
+    path: /{index}/_analyze
+    method: POST
+    parameters:
+      index: example-phone
+    request:
+      payload:
+        text: +41 60 555 12 34
+    response:
+      status: 200
+      payload:
+        tokens:
+          - token: +41 60 555 12 34
```

Comment:
from the action run (on my fork):
```
            FAILED  RESPONSE PAYLOAD BODY (expected tokens[0].token='+41 60 555 12 34', got '41', missing tokens[1].token='6055512')
```

there's no guaranteed order of tokens by the plugin (because it doesn't matter). i now used the order in which it was returned on my machine and just picked the first two (because the first was 1:1 the same content, so no proof that the analyzer actually works).

is there a way to say that i don't want to test the exact content of the list of the response payload but just want to check if the specified values are present (i.e. incomplete list, with the order being irrelevant)?

Reply:
Does the tokenizer really return them in different order?

If this becomes a problem we can implement something smarter.

Reply:
[the implementation](https://github.com/opensearch-project/OpenSearch/blob/9f790ee1e89063fd38501e64faf1df7109b3f4ec/plugins/analysis-phonenumber/src/main/java/org/opensearch/analysis/phone/PhoneNumberTermTokenizer.java#L93) uses a [`HashSet`](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/HashSet.html) which clearly states
>  It makes no guarantees as to the iteration order of the set; in particular, it does not guarantee that the order will remain constant over time.

the order you see in this PR is the order i got for the first two elements when running OpenSearch on windows x64. as it fails here in the CI this is a clear indication that the order is influenced by outside things. thus the test shouldn't rely on the order

Reply:
i just realised that the test doesn't actually set the `analyzer` - so it's not using the correct analyser, thus returning the wrong tokens 😅🙈

Reply:
with the following request:
```
curl --location --request POST 'http://localhost:9200/example-phone/_analyze' \
--header 'Content-Type: application/json' \
--data-raw '{
  "analyzer" : "phone-ch",
  "text" : "+41 60 555 12 34"
}'
```
against a plain OpenSearch 2.18.0 cluster (security plugin disabled, `analysis-phonenumber` plugin installed, prelude index created; no other changes) i get the following output:
```json
{
    "tokens": [
        {
            "token": "+41 60 555 12 34",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 0
        },
        {
            "token": "6055512",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 1
        },
        {
            "token": "41605551",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 2
        },
        {
            "token": "416055512",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 3
        },
        {
            "token": "6055",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 4
        },
        {
            "token": "41605551234",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 5
        },
        {
            "token": "60555123",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 6
        },
        {
            "token": "41605",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 7
        },
        {
            "token": "4160",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 8
        },
        {
            "token": "4160555",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 9
        },
        {
            "token": "416055",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 10
        },
        {
            "token": "41 60 555 12 34",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 11
        },
        {
            "token": "60555",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 12
        },
        {
            "token": "605551",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 13
        },
        {
            "token": "6",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 14
        },
        {
            "token": "4160555123",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 15
        },
        {
            "token": "605551234",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 16
        },
        {
            "token": "416",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 17
        },
        {
            "token": "60",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 18
        },
        {
            "token": "605",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 19
        },
        {
            "token": "41",
            "start_offset": 0,
            "end_offset": 0,
            "type": "word",
            "position": 20
        }
    ]
}
```

yet the test still fails:
```
FAILED  RESPONSE PAYLOAD BODY (missing tokens[1].token='6055512')
```

(see [the test run on my fork](https://github.com/rursprung/opensearch-api-specification/actions/runs/11746394228/job/32726049687))

i'm not sure why it doesn't find this - that should be the 2nd token as per the list? 🤔

Reply:
Because what you get back is not a number, but a string. So quote the value.